nomenclature/region_mapping_models.py

```diff
@@ -0,0 +1,103 @@
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+from collections import Counter
+
+import yaml
+from jsonschema import validate
+from pydantic import BaseModel, validator, root_validator
+
+here = Path(__file__).parent.absolute()
+
+
+class NativeRegion(BaseModel):
+    name: str
+    rename: Optional[str]
+
+    @property
+    def target_native_region(self):
+        return self.rename if self.rename is not None else self.name
+
+
+class CommonRegion(BaseModel):
+    name: str
+    constituent_regions: List[NativeRegion]
+
+
+class RegionAggregationMapping(BaseModel):
+    model: str
+    native_regions: Optional[List[NativeRegion]]
+    common_regions: Optional[List[CommonRegion]]
+
+    @validator("native_regions")
+    def validate_native_regions(cls, v):
+        target_names = [nr.target_native_region for nr in v]
+        doubles = [item for item, count in Counter(target_names).items() if count > 1]
+        if doubles:
+            raise ValueError(
+                f"Two or more native regions share the same name: {doubles}"
+            )
+        return v
+
+    @validator("common_regions")
+    def validate_common_regions(cls, v):
+        names = [cr.name for cr in v]
+        doubles = [item for item, count in Counter(names).items() if count > 1]
+        if doubles:
+            raise ValueError(
+                f"Duplicated aggregation mapping to common regions: {doubles}"
+            )
+        return v
+
+    @root_validator()
+    def check_illegal_renaming(cls, values):
+        """Check if any renaming overlaps with common regions"""
+        if values.get("native_regions") is None or values.get("common_regions") is None:
+            return values
+        native_region_names = {
+            nr.target_native_region for nr in values["native_regions"]
+        }
+        common_region_names = {cr.name for cr in values["common_regions"]}
+        overlap = list(native_region_names & common_region_names)
+        if not overlap:
+            return values
+        raise ValueError(
+            "Conflict between (renamed) native regions and aggregation mapping"
+            f" to common regions: {overlap}"
+        )
```

Comment:
Suggest to change the order for more logical reading...
```suggestion
        if overlap:
            raise ValueError(
                "Conflict between (renamed) native regions and aggregation mapping"
                f" to common regions: {overlap}"
            )
        return values
```